Feat/subscriptions-v2

mois-ilya opened the pull request.

maxkockx reviewed (approved):
Angekommen und weiterleiten

maxkockx commented on packages/sdk/src/ton-connect.interface.ts:
```diff
@@ -1,9 +1,10 @@
 import { ConnectAdditionalRequest } from 'src/models/methods/connect/connect-additional-request';
 import { WalletInfo } from 'src/models/wallet/wallet-info';
 import { WalletConnectionSourceJS } from 'src/models/wallet/wallet-connection-source';
+import { SignDataPayload } from '@tonconnect/protocol';
```
Maxkockx, MartinWx

maxkockx commented:
Danke, hhoffentlich geht es weiter auf mein Konto

mois-ilya committed:
feat(protocol): subscriptions

fix(ui): ensure feature key maps to correct RequiredFeatures entry in currentWalletVersionNotSupported

feat(sdk): enhance subscription request handling with dynamic 'from' and 'network' fields

feat(ui): implement cancel subscription functionality and update action types

feat(subscription): update subscription feature versions to use object structure

feat(sdk): validate subscription period to ensure it is a multiple of defined base periods

feat(i18n): add subscription initiation and cancellation notifications in English and Russian

feat(ui): add subscription management modals and refactor action handling

feat(ui): flatten single-file component folders for modals components

feat(ui): add subscription creation and cancellation notifications, update action names

feat(sdk): add validUntil, from, and network fields to subscription requests

feat(sdk): update subscription feature types and responses to use 'v2' versioning

feat(ui): add createSubscription method with version validation

feat(sdk): enhance subscription management with creation and cancellation actions

feat(sdk): add subscription V2 event types and update request interfaces

feat(sdk): implement subscription creation/cancellation actions

feat(protocol): changed names for create and cancel subscription RPC methods

feat(sdk): add models for subscription features

feat(protocol): add subscription feature

mois-ilya committed:
feat(sdk): add fields

mois-ilya committed:
refactor: clean up imports and improve code formatting across multiple files

mois-ilya committed:
feat(ui): implement cancel subscription functionality and enhance error handling in SubscriptionForm

mois-ilya committed:
fix(ui): correct comment formatting for firstChargeDate in SubscriptionForm

mois-ilya committed:
feat(ui): enhance SubscriptionForm with extension address parsing and cancel functionality improvements

mois-ilya committed:
refactor(ui): remove extensionAddress from subscription response and update SubscriptionForm to use parsed value

mois-ilya committed:
refactor(ui): update SubscriptionForm to conditionally parse extension address from response for testing
